arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -303,18 +303,14 @@ internal fun <T> Iterable<T>.collectionSizeOrDefault(default: Int): Int =
 public inline fun <E, A, B> Iterable<A>.traverseEither(f: (A) -> Either<E, B>): Either<E, List<B>> =
   traverse(f)
 
+@Deprecated(
+  "Traverse for Either is being deprecated in favor of Either DSL + Iterable.map.\n$NicheAPI",
+  ReplaceWith("either<E, List<B>> { this.map { f(it).bind() } }", "arrow.core.raise.either")
```

Comment:
If we introduce `bindAll`, maybe a better replacement would be the following:

```suggestion
  ReplaceWith("either<E, List<B>> { this.map(f).bindAll() }", "arrow.core.raise.either")
```

Reply:
(and the same for the implementation)

Reply:
This is actually wrong, because with `map(f).bindAll()` we make two runs over the list, which changes the side effects which are run.